optimum/onnxruntime/modeling_ort.py

```diff
@@ -111,36 +118,55 @@ def device(self) -> torch.device:
         return self._device
 
     @device.setter
-    def device(self, value):
+    def device(self, value: torch.device):
         self._device = value
 
-    def to(self, device):
+    def to(self, device: torch.device):
         """
         Changes the ONNX Runtime provider according to the device.
         """
         self.device = device
         provider = get_provider_for_device(self.device)
         self.model.set_providers([provider])
+        self.providers = self.model.get_providers()
         return self
 
     def forward(self, *args, **kwargs):
         raise NotImplementedError
 
     @staticmethod
-    def load_model(path: Union[str, Path], provider=None):
+    def load_model(
+        path: Union[str, Path],
+        provider: Optional[Union[str, List[str]]] = "CPUExecutionProvider",
+        session_options: Optional[ort.SessionOptions] = None,
+        **kwargs
+    ):
         """
         Loads an ONNX Inference session with a given provider. Default provider is `CPUExecutionProvider` to match the default behaviour in PyTorch/TensorFlow/JAX.
 
         Arguments:
             path (`str` or `Path`):
                 Directory from which to load the model.
-            provider(`str`, *optional*):
-                ONNX Runtime provider to use for loading the model. Defaults to `CPUExecutionProvider`.
+            provider (`str` or `List[str]`, *optional*):
+                ONNX Runtime providers to use for loading the model. This can either be a single provider given as a string (for example
+                `CUDAExecutionProvider`) or a list of execution providers to use in priority order.
+                See https://onnxruntime.ai/docs/execution-providers/ for more details. Defaults to `CPUExecutionProvider`.
+            session_options (`onnxruntime.SessionOptions`, *optional*),:
+                ONNX Runtime session options to use for loading the model. Defaults to `None`.
         """
-        if provider is None:
-            provider = "CPUExecutionProvider"
+        if isinstance(provider, str):
+            providers = [provider]
+        elif isinstance(provider, list):
+            providers = provider
```

Comment:
Because it is a feature from onnxruntime. But maybe not used so much, I am not sure, so if you prefer to support only a single execution provider I can change that, let me know!

Reply:
Currently we only support cpu and cuda so I don't think it makes a difference to give the possibility to give a list of execution providers with a priority order, so for clarity I would keep it as a string for the moment